cpp/cmake_modules/DefineOptions.cmake

```diff
@@ -96,6 +96,12 @@ if("${CMAKE_SOURCE_DIR}" STREQUAL "${CMAKE_CURRENT_SOURCE_DIR}")
   define_option(ARROW_BUILD_BENCHMARKS "Build the Arrow micro benchmarks, default OFF"
                 OFF)
 
+  # Reference benchmarks are used to compare to naive implementation, or
+  # discover various hardware limits.
+  define_option(ARROW_BUILD_BENCHMARKS_REFERENCE
```

Comment:
Really just a nit, but `ARROW_WITH_REFERENCE_BENCHMARKS` might be a better choice.

Reply:
Well, there's `ARROW_BUILD_BENCHMARKS` already. The convention seems to be `ARROW_BUILD_xxx` to enable the `xxx` target, and `ARROW_WITH_yyy` to enable support for the `yyy` external library.

Reply:
do you mean `ARROW_BUILD_REFERENCE_BENCHMARKS`?